### DISCUSSION TITLE
Can somebody explain Hscript for me?

### DISCUSSION THREAD

**Discussion by 40U538072** [CodenameCrew/CodenameEngine#54] [General]
I was informed that this engine supports Hscript instead of lua if I am not mistaken. I am more of a lua guy, but Psych Engine has several broken bugs that went on for so long and I hate it, and it supported lua, and someone told me this was a good alternative. Is there any crash course on Hscript or is this the wrong engine? Also, is it better than and easier to work with than lua?

**Comment by Minyatan** [comment]
hscript is kinda just haxe/haxeflixel (with some limitations ofc)
if you dont know haxeflixel, i recommend starting here ^^ https://haxeflixel.com/documentation/getting-started/

**Reply by 40U538072** [reply]
> hscript is kinda just haxe/haxeflixel (with some limitations ofc) if you dont know haxeflixel, i recommend starting here ^^ https://haxeflixel.com/documentation/getting-started/

I see, but this is the installation page. I was wondering how to learn this hscript. I know lua but Psych Engine is buggy and is the only engine that uses lua. I was considering learning hscript since I heard this engine is allegedly better. With the page you sent me, I can install it. I will need to learn coding with hscript.

**Reply by Minyatan** [reply]
> > hscript is kinda just haxe/haxeflixel (with some limitations ofc) if you dont know haxeflixel, i recommend starting here ^^ https://haxeflixel.com/documentation/getting-started/
> 
> I see, but this is the installation page. I was wondering how to learn this hscript. I know lua but Psych Engine is buggy and is the only engine that uses lua. I was considering learning hscript since I heard this engine is allegedly better. With the page you sent me, I can install it. I will need to learn coding with hscript.

i dont get what you mean, if you learn haxeflixel you'll be learning hscript too. they are the same language basically

also, i dont really know what you mean by "explain hscript"? like explain the basic fundamentals?

**Reply by 40U538072** [reply]
> > > hscript is kinda just haxe/haxeflixel (with some limitations ofc) if you dont know haxeflixel, i recommend starting here ^^ https://haxeflixel.com/documentation/getting-started/
> > 
> > 
> > I see, but this is the installation page. I was wondering how to learn this hscript. I know lua but Psych Engine is buggy and is the only engine that uses lua. I was considering learning hscript since I heard this engine is allegedly better. With the page you sent me, I can install it. I will need to learn coding with hscript.
> 
> i dont get what you mean, if you learn haxeflixel you'll be learning hscript too. they are the same language basically
> 
> also, i dont really know what you mean by "explain hscript"? like explain the basic fundamentals?

I really don't understand the whole haxeflixel thing. It doesn't look like anything simple for me to follow. I don't know if a download is even possible. I was really asking for the basic fundamentals because I can't really find any useful tutorials on it.

**Reply by LeoFour4L** [reply]
You do pretty much the same thing as in Psych Engine's lua, but instead of functions like setProperty("boyfriend.x", 50), you just type PlayState.boyfriend.x = 50; There's also switch(stuff) element with cases, but it's not only limited to hscript (it's present in other programming languages I mean), which is (as Foxel said) haxeflixel but with certain limitations. If you want to work with hscript, you will have to understand haxeflixel. For example, for tweens you use FlxTween. For timers you use FlxTimer, and yada yada etc.
Believe it or not, hscript will become easy for you once get into it. To get started, you can look for scripts in engine's folder and for scripts on Codename Engine's Discord server. I know, there's not as much as the amount of scripts for PE, but they'll certainly help you to get a clearer picture of how things work.

**Reply by 40U538072** [reply]
> You do pretty much the same thing as in Psych Engine's lua, but instead of functions like setProperty("boyfriend.x", 50), you just type PlayState.boyfriend.x = 50; There's also switch(stuff) element with cases, but it's not only limited to hscript (it's present in other programming languages I mean), which is (as Foxel said) haxeflixel but with certain limitations. If you want to work with hscript, you will have to understand haxeflixel. For example, for tweens you use FlxTween. For timers you use FlxTimer, and yada yada etc. Believe it or not, hscript will become easy for you once get into it. To get started, you can look for scripts in engine's folder and for scripts on Codename Engine's Discord server. I know, there's not as much as the amount of scripts for PE, but they'll certainly help you to get a clearer picture of how things work.

I will take a look at it. Where can I find the Discord server?

**Reply by LeoFour4L** [reply]
> > You do pretty much the same thing as in Psych Engine's lua, but instead of functions like setProperty("boyfriend.x", 50), you just type PlayState.boyfriend.x = 50; There's also switch(stuff) element with cases, but it's not only limited to hscript (it's present in other programming languages I mean), which is (as Foxel said) haxeflixel but with certain limitations. If you want to work with hscript, you will have to understand haxeflixel. For example, for tweens you use FlxTween. For timers you use FlxTimer, and yada yada etc. Believe it or not, hscript will become easy for you once get into it. To get started, you can look for scripts in engine's folder and for scripts on Codename Engine's Discord server. I know, there's not as much as the amount of scripts for PE, but they'll certainly help you to get a clearer picture of how things work.
> 
> I will take a look at it. Where can I find the Discord server?

You can find it in the credits of YoshiCrafterEngine. (https://github.com/YoshiCrafter29/YoshiCrafterEngine/blob/9d6b4209a855b5b27d52a20d3ea2b1f4c3a51171/assets/preload/credits.json)

**Comment by biomseed** [comment]
Haxe except it's a scripting language. That's literally it, i wish there was something like this for other languages.

**Reply by 40U538072** [reply]
Isn't lua also a scripting language?

**Reply by biomseed** [reply]
> Isn't lua also a scripting language?

Yes, but hscript allows you to bring haxe classes and functions over to hscript, allowing you to be able to do everything you can do in haxe. For example you can use FlxG, Playstate, FlxSprite, and more from hscript without having to modify the source code. This has a huge advantage over lua because it is easier to implement into an engine, and basically has no limitations except for not being able to import libraries without importing them into the engine as well, but that's not something you really need.
